base/install-functions.sh

```diff
@@ -1,3 +1,11 @@
+get_module_path() {
+    for module; do
+        if [ -n "$module" ]; then
+            grep -E "^$module=" $EPICS_RELEASE_FILE
+        fi
+    done
```

Comment:
Ok, I'm now a bit confused... This shouldn't be working, and yet the build succeeds.

You are passing the arguments as a quoted list; that means your `for` loop runs only once, and searches for all the modules in a single grep, and shouldn't find any matches... Example from CI run:

```
2024-01-24T20:38:45.4438379Z #19 249.2 + grep -E '^
2024-01-24T20:38:45.4438645Z #19 249.2 EPICS_BASE
2024-01-24T20:38:45.4438981Z #19 249.2 ASYN
2024-01-24T20:38:45.4439209Z #19 249.2 CALC
2024-01-24T20:38:45.4439485Z #19 249.2 =' /opt/epics/RELEASE
```

For it to work as you expected, do `for module in $@;`.

Since we are making changes here, I think we should also error out if the output is empty, as will be the case for the wrong module name for sequencer (SEQ) when building CALC (this PR should then also fix that in a first commit).

Reply:
> Ok, I'm now a bit confused... This shouldn't be working, and yet the build succeeds.

Yeah, it should break with no dependencies defined.

> Example from CI run

Nice to see that `-x` turned out to be really useful in the end. ;)

> For it to work as you expected, do for module in $@;.

Makes sense. I'll update it.

> I think we should also error out if the output is empty

That would mean we couldn't use it for EPICS base, right?

> this PR should then also fix that in a first commit

Good point. I did notice this in other context, but I forgot about that when making these changes.

Reply:
> That would mean we couldn't use it for EPICS base, right?

I've checked this and it is not the case. We'll be passing an empty argument, which won't even trigger the `grep` search, successfully passing by.

> I think we should also error out if the output is empty

That's currently the case. Since we set `-e` shell option, `grep` can't return non-zero, which is what happens when it doesn't find a match.

Reply:
> That's currently the case. Since we set -e shell option, grep can't return non-zero, which is what happens when it doesn't find a match.

Please document it in the commit message :)